### ISSUE TITLE
Geany 1.33 AutoSave FilesOpenList, using Debian Buster 10 in Oracle VirtualBox inside Windows 10 PC

### ISSUE THREAD

**Issue by SymbReprUnlim** [geany/geany#3502]
Issues/Statement of Issues to be hopefully Solved:

1a) Linux Terminal commands sudo apt update and sudo apt full-upgrade never update Geany 1.33 to its current version of 1.38. Thus Geany 1.33 remains apparently the "latest available" version of Geany within a Windows PC running Oracle Virtual Box Debian Buster 10; and thus a 32-bits OS in Debian itself, NOT 64-bits, although the host Windows PC OS is 64 bits and some documentation indicates support for Geany 32-bits is either gone away or going away; hence Geany 1.38 appears 64-bit usage only.

1b) In "EZ" terms or .sh (Linux batch command file) usage, is there an easy or user-friendly way to update Geany to a more current version of Geany, like 1.38/1.37 ... or a last known 32-bit compliant/allowable Linux/Debian Geany version? The generalized/documented/wiki instructions: 1) download these libraries first, 2) make sure these Make/Compile/Build commands get executed/run and then 3) move X resultant Geany production files to correct folders - is not a simple, straightforward process; but it shouldn't be impossible either.

2) Triggering and PRIMARY Issue here (Save Actions Plugin): From time to time, the RPi/Debian/Windows 10 VBox PC, for all intents and purposes, reboots; as if losing power; for off-hours updates by Windows and or a virus protection program. Upon full boot-up and VBox boot into Debian, the "newly" rebooted RPi completely loses its previous configuration. Especially troubling is that the list of formerly opened list of Geany files (on average 25-35 files) is totally gone, entirely wiped out. Multiple python program files must all therefore be manually re-opened inside Geany 1.33. It's not so much a problem that files don't get saved, since user vigilance and Geany's "red highlighting" unsaved files helps - Save those files prior to off-hours, walk-away, nightly major OS updates.  Under Geany's Plug-Ins, there is a Tools-Save Actions, but under Edit-PlugIn Preferences - only 2 options show up a) Save only current open file and b) Save all open files. I don't need either of those options, but rather a "Save Geany 'Currently Loaded Files' List."  Again, not saving the actual files themselves to disk, but rather saving Geany's Currently Opened Files Configuration (file names, their exact sequential Geany tabs order and with the last user recently opened/used file "on top-actively visible"), so to speak. Is there a way to have Geany do this?

In short, I am a well above average in experience programmer(mostly Python, C#, a little C), but one who also lacks specific GTK knowledge and who lacks specific knowledge about attempting to Make/Compile/Build/Install-correctly Geany itself. So I'm offering to help at: A) attempting to make Geany 1.38 (or its Save Actions Plugin) work on Debian Buster 10 32-bits with an Auto-Save Opened Files Configuration OR B) allowing Geany 1.33 to Auto-Save its Opened Files Configuration to Disk, as part of the File Actions Geany Plug-In.

Any thoughts, helpful suggestions, or proposed/implemented solutions?

Thank You!

**Comment by eht16** [comment]
> 1a) Linux Terminal commands sudo apt update and sudo apt full-upgrade never update Geany 1.33 to its current version of 1.38. Thus Geany 1.33 remains apparently the "latest available" version of Geany within a Windows PC running Oracle Virtual Box Debian Buster 10; and thus a 32-bits OS in Debian itself, NOT 64-bits, although the host Windows PC OS is 64 bits and some documentation indicates support for Geany 32-bits is either gone away or going away; hence Geany 1.38 appears 64-bit usage only.

This is out of scope of the Geany project. You are asking why the Linux distribution you are using provides only an outdated version of a particular software package. This is a question for the Debian project. I would recommend to read about Linux distributions, how they work and especially how package managing works.
To solve your problem, you have basically two options:
- update your Linux distribution: Debian Buster in several years old and never will get any more updates, so your expectation will never be fulfilled
- compile Geany yourself on the old system, see below


> 1b) In "EZ" terms or .sh (Linux batch command file) usage, is there an easy or user-friendly way to update Geany to a more current version of Geany, like 1.38/1.37 ... or a last known 32-bit compliant/allowable Linux/Debian Geany version? The generalized/documented/wiki instructions: 1) download these libraries first, 2) make sure these Make/Compile/Build commands get executed/run and then 3) move X resultant Geany production files to correct folders - is not a simple, straightforward process; but it shouldn't be impossible either.

I don't know what "EZ" is. Compiling Geany from sources using the instructions in the README and manual should not be too hard for an "well above average in experience programmer". What exactly do you expect here? We, as a group of volunteers who spend their spare time on the development of Geany, cannot effort to provide ready-to-use binary packages for arbitary operating systems, distributions, hardware and all the combinations of those.


>     2. Triggering and PRIMARY Issue here (Save Actions Plugin): From time to time, the RPi/Debian/Windows 10 VBox PC, for all intents and purposes, reboots; as if losing power; for off-hours updates by Windows and or a virus protection program. Upon full boot-up and VBox boot into Debian, the "newly" rebooted RPi completely loses its previous configuration. Especially troubling is that the list of formerly opened list of Geany files (on average 25-35 files) is totally gone, entirely wiped out. Multiple python program files must all therefore be manually re-opened inside Geany 1.33. It's not so much a problem that files don't get saved, since user vigilance and Geany's "red highlighting" unsaved files helps - Save those files prior to off-hours, walk-away, nightly major OS updates.  Under Geany's Plug-Ins, there is a Tools-Save Actions, but under Edit-PlugIn Preferences - only 2 options show up a) Save only current open file and b) Save all open files. I don't need either of those options, but rather a "Save Geany 'Currently Loaded Files' List."  Again, not saving the actual files themselves to disk, but rather saving Geany's Currently Opened Files Configuration (file names, their exact sequential Geany tabs order and with the last user recently opened/used file "on top-actively visible"), so to speak. Is there a way to have Geany do this?

This is the good news after all: recent Geany versions (definetely 1.38, maybe also 1.37 already) save the list of open files, we call it "session", automatically whenever a file is created, opened or closed. So, once you will use a recent version of Geany, the requested feature is already available.

**Comment by SymbReprUnlim** [comment]
Yes, all of us are very busy, me included. I appreciate all the time you and any other Geany volunteers have put into developing Geany, including the time you have taken to at least read my perspective and same-day respond to it.

However, in short, your answer seems to be: Follow the instructions in the Readme and manual. Go away experienced programmer, Geany no longer supports 32 bit Operating Systems. Don't bother us with keeping 32-bit Operating Systems up to date. We only look forward, but out of necessity and our limited available time limits - we (the Geany Programmers) are thus time constrained, inadequate labor-supply faced.

Duly noted.  Geany programmers have also consciously decided, based on time constraints to stop supporting 32- bit OS's. I understand that, yet thank you for clarifying.

"8 Linux Distros That Still Support 32-Bit Architecture", from 4/15/2023 ref. https://www.makeuseof.com/linux-distros-with-32-bit-support/   and "Top 10 32-Bit Linux Distributions in 2023" from 1/31/2023 ref. https://www.debugpoint.com/32-bit-linux-distributions/ 

Unfortunately, that means I and other 32-bit OS users must evaluate using other IDE, non-Geany programming alternatives. And the last stated major Geany release for version 1.38 occurred in October, 2021 - about 1.5 YEARS ago. To this experienced programmer, facts would seem to imply on a surface level at least:, Geany is not necessarily a viable software package utility going forward for 32-bit systems. Lack of support for them going forward sounds like an issue.

Converting a 32-bit system to a 64-bit OS isn't necessarily possible in many cases.  You have also NOT reassured me that it is even possible to use Geany 1.38 on a 32-bit Linux/Debian OS, once all requisite steps get followed to perfection.

For what it's worth, Geany release 1.33 has been, for the most part a very reliable, helpful IDE on Debian 32-bit in the past. And again, thank you very, very much for your time and personal+professional involvement. I get it.

**Comment by elextr** [comment]
> Unfortunately, that means I and other 32-bit OS users must evaluate using other IDE, non-Geany programming alternatives.

See the end of https://github.com/geany/geany/issues/3063#issuecomment-1539252949

Anyway, why are you ranting at the Geany project, its your distro __Debian__ that decides not to update from 1.33, not the Geany project.  AFAIK there is nothing that will prevent Geany 1.38 being built for a 32 bit OS, but its __Debian__ that needs to do it if you are using a __Debian__ distro.

**Comment by SymbReprUnlim** [comment]
Examining versus ranting are two entirely separate & distinct mental processes. Thank you very much for at least pointing out, well scrutinizing, that Geany 1.38 CAN HOPEFULLY be used with a Debian distro. A bit of validation/optimism makes a task easier, more likely worth implementing and adapting to. Muchos Gracias Elextr!, from a 32-bit but worthwhile OS implementer for many, many very high precision, numeric, multi-tasking, and intensive projects. Historical note: 8-bit OS's stopped being used in the early 1980's; 16-bit apps can still, proof of concept anyway, be run on Windows 10 64-bit OS PC's, ref. Microsoft's website 9/15/2020 https://techcommunity.microsoft.com/t5/modern-work-app-consult-blog/running-16-bit-applications-on-windows-10-64-bit/ba-p/1671418

**Comment by elextr** [comment]
>  that Geany 1.38 CAN HOPEFULLY be used with a Debian distro.

LMGTFY  https://packages.debian.org/search?keywords=geany

But note, that page shows (as @eht16 pointed out) Debian has decided that they will not upgrade the version of Geany (well, most applications IIUC) that is part of Buster.  To get a newer version of Geany you need to use a newer version of Debian or build it yourself.

Just to clarify the concept of distributions, since you are running in a VM on Windows and Windows does not have the concept of distributions.  A distribution (distro) is a collection of Linux plus application software put together by an independent organisation, Debian, Arch, Ubuntu, Fedora etc.  Even Microsoft has its own Linux distro for WSL2.  Those organisations and their contributors choose and package the software they provide and the policy of upgrading versions that they use.  The upgrade policy can range from "everyday" for rolling ones like Arch to "never" for stable versions like Debian Buster.  Application projects like Geany do not decide what Geany version is in any version of any particular distro.

**Comment by SymbReprUnlim** [comment]
Thanks for the update!

I've looked more closely at Geany's Manual at https://www.geany.org/manual/current/index.html  and specifically under https://www.geany.org/manual/current/index.html#installation
For most people, me included, if the software command isn't a simple 
$ pip install geany 1.38 (like a Python program/lib install) OR MORE LIKELY:
$ sudo apt install geany 1.38  (from a Terminal)
we, the NON-erudite programmers sub-class or "Microsoft previously handheld use Visual Studio auto-make-compile-build-link/etc." subset (and there are plenty of us) - we will fail miserably at any non-straightforward software install.
And thus the Geany install process "appears" to be well, well beyond an easy, normal software setup, for most people.

So I did a Google Search on the terms: linux geany 1.38 install
The following link appeared: How to Install Geany IDE 1.38 via PPA in Ubuntu 20.04 / 18.04 / 21.10  https://ubuntuhandbook.org/index.php/2021/12/install-geany-ppa-ubuntu/
and  https://launchpad.net/~geany-dev/+archive/ubuntu/ppa
It's Ubuntu, but not Debian - but the two OS's are at least somewhat similar cousins and what works for Ubuntu, frequently but not always - works for Debian.  That Ubuntu link then led me back to https://www.geany.org/download/third-party/

BUT the Ubuntu PPA is apparently 130 weeks old or 2.5 years out of date and isn't a sure install bet anyway for Debian even if it were up to date. Furthermore, I seriously doubt Debian will have a successful Terminal install using
$ sudo apt remove geany 1.33
$ sudo apt install geany 1.38
precisely because the requisite files/libs/modules will not already be installed and available on Buster/Debian 10 32-bit pc.

I am trying to communicate my view, correct or incorrect, on what an average programmer is likely to see when he/she looks at Geany's 1.38 Manual and might start to attempt to install Geany, then likely have that install process take a week and finally jump into the arms of Microsoft's Visual Studio Code, only to be hoodwinked out of another week's Linux IDE install take-down time.

You guys are familiar with Geany's total technical install subparts and multiple pre-existing library requisites, we the simple-minded "sudo apt installers" are not. 

How to Install Software on Linux – Beginners Guide
from August 2, 2022:
Number 1 Method: Use the Package Manager!!!
A package manager is a software tool that allows users to install, upgrade, search, download and remove software packages on an operating system. https://linuxopsys.com/topics/install-software-on-linux
And as of right now, this 3-4 years Linux programmer has not yet found a package manager tool to install Geany 1.38 on Debian 10 Buster 32-bit OS, forgive me. Again, thank you to everyone who has contributed to this thread thus far.

**Comment by elextr** [comment]
> For most people, me included, if the software command isn't a simple
> $ pip install geany 1.38 (like a Python program/lib install) OR MORE LIKELY:
> $ sudo apt install geany 1.38 (from a Terminal)

Geany is a C program, it needs to be compiled before it can run, and to do that from source needs the dependencies and build tools to be available because the source does not know how to install them because it is not targeted to one distro.

The simplified install is part of what what distros can provide (`apt` is a command developed by Debian, other distros have other similar commands).  Distros can do this because they know the specifics of the system they are installing on, can know and install dependencies, and can store executables in the online repositories so no compile is needed.  But you are limited to the distros provided versions, which is where we started ...

Ubuntu is based on Debian, but Ubuntu packages probably will not install on Debian, especially on an old distro version like buster, library versions are not identical and distros are not identical, so software needs to be compiled for the specific distro and version.

As my previous link listed, Geany 1.38 is available on newer Debians.  That is probably more likely to work if installed on Buster, but no guarantees since newer builds are likely 64 bit systems.

Just a note, the Ubuntu PPA is up to date, a pandemic, and spawning of children, and other reasons (see #3499) have meant that it is indeed a long time since last official release (1.38) of Geany.  We are aware of that.

Open source allows anyone to make a distro to suit their useage or to include their favourite software, and even to clone another distro and only change a few parts, thats why there are so many distros https://en.wikipedia.org/wiki/List_of_Linux_distributions.

But because each distro has different software management tools and software versions and techniques there is no way the Geany README can list them all, or keep such a list up to date.

> Number 1 Method: Use the Package Manager!!!
> A package manager is a software tool that allows users to install, upgrade, search, download and remove software packages on an operating system. https://linuxopsys.com/topics/install-software-on-linux

Thats the best advice for nearly everyone.  The simplest should be the distro specific user friendly GUI tool, eg Software Manager on my Mint that tries to be like Google Play/Apple App Store.  Then the lower level commands such as `apt` or its GUI version `synaptic` which are package managers, but as noted they can only install what is present in their repositories, which again comes back to where we started ...

The Geany README and INSTALL instructions relate to building/installing from source, either a tarball or a clone of the git repository.  But it is not a primer on building C code, build tools, installing dependencies and tools and other items that are generic to building C code.

**Comment by SymbReprUnlim** [comment]
Hi elextr and anyone else who might choose to respond, hypothesize and or contribute to this thread. I appreciate (elextr) your response very much AND your responses is/are factual, useful information - Emphasis Added!!  But at the same time you are not informing me beyond the boundary of my previously known information, my personal pre-existing knowledge base so to speak.

On the one hand, I cannot change how Debian aka Raspbian/Raspberry Pi chooses to allow its "decided upon" package manager "apt" to accumulate its allowable software versions, in the instant case here Geany 1.33, with A,B,C,D etc... dependent/requisite libraries in place. I really do not like fighting organizations (Debian in this case) to affect forward change - which I would likely have to do -to get Geany 1.38 as the basis for an actual Debian respository/repositories.

On the other hand to seemingly satisfy Geany's Developers current requirements, this Geany user doesn't want to be - likely multiple, repetitive times - making, compiling, debugging, linking and copying files, installing requisite libraries ... manually - which by going step by step through the Geany 1.38 manual, you guys are suggesting I need to do to get Geany 1.38 working on this particular 32-bit Debian Buster 10 OS. And all to be lost/replaced upon a newer version of Geany becoming available.

This means both proposed solution paths (Debian apt AND Geany Manual) are either impractical and or overly time consuming for this end-user and anyone else similarly faced, for a valid Geany 1.38 install.

Proposed Solution Path 1: My particular forte on Linux is Python (it is a popular programming language these days), including issuing OS commands from a Python program and returning the results of those OS commands to Python - for evaluation and or further processing. I should in theory, for example, be able to auto-compile a C program by issuing a Python command (from a Geany edited+compiled Python pgm), and retrieve results on whether or not the Compile successfully completed and say output filename X.exe with a new creation date just successfully occurred, or not. If yes, automatically move on to next local Geany version 1.38 creation step. If not, find out why, attempt to "auto remedy" the situation, and repeat Compile/Make Geany 1.38 .Python program ...

Proposed Solution Path 2: Use a "different" or "modified" Package Manager (not Debian repository "apt" per se, as is, standard) software tool - to attempt to get Geany 1.38 installed. For example: 5 Best Linux Package Managers for Linux Newbies:
1. DPKG – Debian Package Management System / APT (Advanced Packaging Tool) / Aptitude Package Manager or SYNAPTIC Package Manager - Synaptic is a GUI package management tool for APT based on GTK+ (like Geany) and it works fine for users who may not want to get their hands dirty on a command line. It implements the same features as the apt-get command line tool.
2. RPM (Red Hat Package Manager) / YUM (Yellowdog Updater, Modified) / DNF – Dandified Yum
3. Pacman Package Manager – Arch Linux
4. Zypper Package Manager – openSUSE
5. Portage Package Manager – Gentoo
Source: https://www.tecmint.com/linux-package-managers/
Again, I am unable to use Debian's "apt" tool "as is" on this machine because "sudo apt update" and "sudo apt full-upgrade" do NOT currently update Geany to 1.38.

Lacking input back from any of you, the Geany Development Team, or anyone else, I will likely first choose the Python solution path, not a standard app Package Manager per se (although Synaptic appears to have attractive attributes re Geany).  I would prefer something that might best work long term. The Python solution is itself potentially problematic for other Debian Geany 1.38 installer-users since there are likely Python libraries installed on this machine that other Debian 32-bit OS's might not have installed.

In summary, the task faced by this hopeful Geany 1.38 Debian Buster-10 32-bit OS user - is writing or implementing a custom Geany 1.38 Installer/Package Manager that ALSO attempts to locally compile using C / C+ / GTK requisite libraries - while also allowing for auto-downloading requisite libraries - as well as cleaning up after completion - and eventually uninstalling Geany 1.38 for a newer version or user dissatisfaction with Geany 1.38 itself.

Any thoughts, suggestions, preferred solutions path(s), potentially helpful links to pursue, tutorials to check out, etc.?  This is looking more and more like a long-term project, rather than a quick and easy fix/install. And like you guys, I have multiple other projects to pursue...

“Optimism is a strategy for making a better future. Because unless you believe that the future can be better, you are unlikely to step up and take responsibility for making it so.”  - Noam Chomsky, American Linguist

Best Regards

**Comment by elextr** [comment]
> On the other hand to seemingly satisfy Geany's Developers current requirements,

I don't understand what you mean by that.

> I should in theory, for example, be able to auto-compile a C program by issuing a Python command

Indeed, its called `meson`, see Geany's `README`.  This "script" is so complex it has its own project https://mesonbuild.com/.  But even so it can't install the build dependencies and tools needed, because they and their install process are intimately linked to the distro and there are so many of them, although it can check they are present at the right version.

>  Use a "different" or "modified" Package Manager (not Debian repository "apt" per se, as is, standard) software tool - to attempt to get Geany 1.38 installed.

A different package manager won't help, you still need the package for your Debian to be built by "somebody" .  Thats what you are lacking for `apt`, its not some failure of the package manager, its a lack of the specific package built for your specific version of Debian.

> In summary, the task faced by this hopeful Geany 1.38 Debian Buster-10 32-bit OS user ....

Might just be easier to build from source tarball. :smile: 

As I said I am not aware of anything that will prevent Geany 1.38 from building on a i386 platform ... if the required versions of libraries are available.

> Any thoughts, suggestions, preferred solutions path(s), potentially helpful links to pursue, tutorials to check out, etc.?

Not really.

Ultimately it comes back to the fact that almost the whole Linux ecosystem is built on volunteer effort, if "somebody" does something and shares it then it will be available, otherwise you need to do it yourself or persuade/pay someone to do it for you.  At least the ability to do that is guaranteed by the availability of source code.

Also support for Buster will end in a little over a years time IIUC, maybe its time to start considering upgrades.

As a final note, I have assumed you need to run Geany in your VM because the files are not visible in Windows, otherwise you could just download the Geany Windows 1.38 installer.

Closing because the conversation has progressed well beyond the scope of Geany.

**Comment by SymbReprUnlim** [comment]
SRU: On the other hand to seemingly satisfy Geany's Developers current requirements,
E: I don't understand what you mean by that.
My response: You have essentially told me to upgrade to Debian 64-bit and 32-bit OS's are not supported by Geany's Development team.

The Python callable C compiler that seemed promising is: SCons is an Open Source software construction tool which orchestrates the construction of software (and other tangible products such as documentation files) by determining which component pieces must be built or rebuilt and invoking the necessary commands to build them. ref. https://pypi.org/project/SCons/

I'm leaving the issue open because you have not resolved allowing 32-bit OS's to be currently supported by Geany 1.38.

**Comment by SymbReprUnlim** [comment]
Actually, users, such as this one, do not seem to have the authority to re-open an issue. So be it. Not my problem. I'll find a different solution path. Thank you for your time.

**Comment by elextr** [comment]
I said:

> As I said I am not aware of anything that will prevent Geany 1.38 from building on a i386 platform ... if the required versions of libraries are available.

Your reply:

> You have essentially told me to upgrade to Debian 64-bit and 32-bit OS's are not supported by Geany's Development team.

incorrectly interprets that as the opposite.

In fact the [Debian package page for Geany](https://packages.debian.org/search?keywords=geany) I linked above shows Debian testing and unstable (ie the bleeding edge of Debian) both have have an i386 architecture package for Geany 1.38.

So at least on Debian it seems that Geany builds on a 32 bit architecture and 32 bit versions of the relevant libraries are available, but that is on versions of Debian newer than Buster.  

Again I will repeat __Debian__ decides what level of support it gives different existing versions, not Geany.  And Debian has decided that Buster is only getting security upgrades and bugfixes (until July 2024), not newer versions of software.  Hence my suggestion to upgrade, I never said to 64 bit.

So it appears there is nothing actionable for Geany, hence this issue is closed.

**Comment by SymbReprUnlim** [comment]
And again, I will repeat that as a very happy Debian 32-bit OS user, Geany 1.38 is not installable as is, and I don't care to fight Debian, nor you + Geany Developers. And again, for the record, going through a manual install of Geany, compiling, linking, etc. - is just not worth a user's time, headaches, manual GTK lib downloads, etc. The end result of both of us going back and forth like this is: The vast majority of Debian 32-bit users are never going to move beyond Geany 1.33. You or a compadre have explicitly not supported Geany 1.38 on Debian 32-bit OSs.

This thread may be closed, but the issues are far from resolved.

**Comment by SymbReprUnlim** [comment]
AND apparently Geany 1.38 is not made easily available to Debian 64-bit OSs.

The commands
$ sudo apt update
$sudo apt full-upgrade
FAIL to upgrade Geany 1.37.1 (a version from June 2021 or so, 2 years ago) to Geany 1.38.

Many, many libraries and software packages, literally hundreds of different software libraries, auto-update using the above 2 terminal commands on Debian, but not Geany.

Whatever software improvements Geany Developers are placing into Geany 1.38 - those particular software "improvements" are not usable to the vast majority of Debian users, both 32-bit AND 64-bit systems. Debian repositories are not reflecting Geany 1.38. Everything else is getting quickly and easily updated on Debian machines, but not Geany.

**Comment by elextr** [comment]
To repeat, the Geany project does not make Debian packages, if a Debian package fails to update/upgrade talk to Debian.  

Repeatedly reporting such things here will just increase your frustration because we will just keep telling you its nothing we can do.

**Comment by SymbReprUnlim** [comment]
I'm not frustrated whatsoever, Geany is not my responsibility. As I perceive it, I'm doing you and Geany a favor. You do seem to fail to understand my point though:

How is it that all these hundreds of other software libraries are getting updated, but not Geany? Debian didn't just "decide" to update all those other libraries and software packages. There is obviously an update process these other libraries and modules via their Developers - went through that Geany is not going through. It's probably not overly difficult to become a part of. Thus:
1) Debian Respository Setup: https://wiki.debian.org/DebianRepository/Setup
2) How Software Producers Can Distribute Their Products (with Debian): https://www.debian.org/doc/manuals/distribute-deb/distribute-deb.html

I'm just the messenger here. I'm not frustrated.

**Comment by elextr** [comment]
> How is it that all these hundreds of other software libraries are getting updated, but not Geany?

I don't know, I will repeat again, the Geany project does not make the Debian packages, telling the Geany project won't help.  You need to tell Debian so they can ask you for whatever is relevant information about exactly what you did, your system configuration etc.  Since the [Debian bugtracker](https://bugs.debian.org/cgi-bin/pkgreport.cgi?dist=unstable;package=geany) does not have any reports of the packages not updating its likely (putting thoughts in the Debian folks heads) that there is something unusual about your setup.

> There is obviously an update process these other libraries and modules via their Developers.

Linux does not work the way Windows (and Macos) does, distros are not like their app stores, developers do not upload distro packages.  Developers make generic software that can be built for many distros, and in the Geany case Windows and Macos.  

The packager makes the package for the distro, and makes sure the way its built matches the distro rules and uses the versions of libraries that the distro provides.  Packagers are specialists in their specific distro, it is impractical for small projects understand the rules for every distro, and to build packages for [all distros](https://en.wikipedia.org/wiki/List_of_Linux_distributions).  

In the case of Debian one of the rules is that only accredited Debian Developers can upload packages after they have checked them to meet the Debian rules, not any Tom, Dick or Harry project.

> I'm just the messenger here. I'm not frustrated.

I'm just trying to point out how you can fix your problem, if a distro package does not exist for a specific distro, or version, or if it fails to upgrade then talk to the distro or build Geany yourself (or have someone do it for you).  Telling Geany is just wasting both our time.
